dig.go

```diff
@@ -320,6 +321,12 @@ func Visualize(c *Container, w io.Writer, opts ...VisualizeOption) error {
 	return _graphTmpl.Execute(w, dg)
 }
 
+// CanVisualizeError returns true if the error is an errVisualizer.
+func CanVisualizeError(err error) bool {
+	_, ok := err.(errVisualizer)
+	return ok
```

Comment:
I think we also want to handle the case where the outer error isn't
visualizable but the inner one  is. So the logic should be more like "is this
thing or anything wrapped by this an errVisualizer"

```
for {
    if errVisualizer { return true }
    if !errCauser { return false }
    err = err.Cause()
}
```

Reply:
Good catch, updated to include all layers of errors.